csrc/ir/internal_base_nodes.cpp

```diff
@@ -1498,6 +1581,22 @@ void TensorDomain::merge(int64_t axis_o, int64_t axis_i) {
   loop_domain_.insert(loop_domain_.begin() + td_outer_pos, merged_id);
 }
 
+// Partition "axis" into component and ragged dimensions. Follow the
+// pattern of TensorDomain::split.
+void TensorDomain::partition(int64_t axis, TensorView* offsets) {
+  NVF_ERROR(nDims() > 0, "Tried to do partition on a 0-dim domain");
+  axis = wrapDim(axis);
+
+  IterDomain* id = this->axis(axis);
+
+  auto [component_id, ragged_id] = RaggedIterDomain::partition(id, offsets);
+
+  // Remove the original axis and insert component and ragged dimensions
+  loop_domain_.erase(loop_domain_.begin() + axis);
+  loop_domain_.insert(loop_domain_.begin() + axis, ragged_id);
+  loop_domain_.insert(loop_domain_.begin() + axis, component_id);
```

Comment:
`vector::erase` returns an iterator pointing to the following element. So you can simplify the code to be

```
auto following = ...;
loop_domain_.insert(following, component_id);
loop_domain_.insert(following, ragged_id);
```
(Note ^^^ component_id is inserted **before** ragged_id)

Reply:
Is this always safe? I don't like the idea of reusing an iterator after modifying containers.

Reply:
I take it back. The first insert is safe but the second insert might be problematic.